### PR TITLE
[4.0] Remove the deprecated code in the Access class

### PR DESCRIPTION
**Pull request by laoneo**
Remove the deprecated code in the Access class.

### CODE REVIEW

File: libraries/src/Access/Access.php [hunk 1]
```diff
@@ -1119,37 +1044,6 @@ public static function getAuthorisedViewLevels($userId)
...
-		if (empty($actions))
-		{
-			return array();
-		}
-		else
-		{
-			return $actions;
-		}
-	}
-
```
Comment (csthomas) [inline libraries/src/Access/Access.php:1152]: First, you have to modify:
```
libraries/src/Form/Rule/RulesRule.php:98:               $elActions = Access::getActions($component, $section);
libraries/src/Form/Field/RulesField.php:162:            $this->actions = Access::getActions($component, $section);
```

**Comment by csthomas** [comment]
There is also a note on line 527.
https://github.com/joomla/joomla-cms/pull/22378/files#diff-692418595f0dc6797e5669cbc3b3870cL527

It will be more difficult.
```
* @note    The non preloading code will be removed in 4.0. All asset rules should use asset preloading.
```

File: libraries/src/Access/Access.php (hunk 1 above)
Comment (laoneo) [reply libraries/src/Access/Access.php:1152]: Thanks, done.

**Comment by laoneo** [comment]
Means, the preload property code should get removed?

**Comment by csthomas** [comment]
IMO `preload` argument in method and the code when preload is false. But it is risky:)

**Comment by laoneo** [comment]
Then better to do it in an isolated pr.

**Comment by wilsonge** [comment]
Thanks!

**Comment by infograf768** [comment]
@laoneo @wilsonge
I think this has killed the display of permissions for the components.

Basically `$xpath` is not working.

**Comment by brianteeman** [comment]
@laoneo @wilsonge this has killed the permissions!!

**Comment by laoneo** [comment]
Please test https://github.com/joomla/joomla-cms/pull/22665 which should fix the rules field.

**Pull request merged**